### PR TITLE
Split `platform::unix` into `platform::x11` and `platform::wayland`

### PR DESCRIPTION
**Pull request by madsmtm**
There were basically no overlap between the platform-specific extensions in these, so it made little sense to keep them in the same file!

Will gladly rename the functions as well, e.g. `wayland_set_csd_theme -> set_csd_theme`?

~Blocked-ish on #2420~

### CODE REVIEW

**Comment by notgull** [comment]
I agree with this decision. Wayland and X11 are very different, so it makes sense to keep them in their own separate modules.

File: src/platform/wayland.rs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+/// Additional methods on [`WindowBuilder`] that are specific to Wayland.
+pub trait WindowBuilderExtWayland {
+    /// Build window with the given application ID.
+    ///
+    /// This should match the `.desktop` file destributed with your program.
+    ///
+    /// For details about application ID conventions, see the
+    /// [Desktop Entry Spec](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#desktop-file-id)
+    fn with_application_id(self, application_id: impl Into<String>) -> Self;
```
Comment (madsmtm) [inline src/platform/wayland.rs:151]: Previously, this was named `with_name`, like it still is in `WindowBuilderExtX11::with_name`, but I felt this `with_application_id` better captures what the meaning actually is!

Comment (kchibisov) [reply src/platform/wayland.rs:151]: This was intentionally and that is to unify X11 and Wayland function to set title. I don't really want to go back to separate methods again. I'm still not sure that I want to split them, given that it doesn't provide that much value.

They where separate before and we had `with_appid` and `with_class`, but there were a request to unify them.

Comment (madsmtm) [reply src/platform/wayland.rs]: Fair enough, I've changed it back

File: src/platform/wayland.rs [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    /// Possible values for env variable are: "dark" and light"
+    fn wayland_set_csd_theme(&self, config: Theme);
+
+    /// Check if the window is ready for drawing
+    ///
+    /// It is a remnant of a previous implementation detail for the
+    /// wayland backend, and is no longer relevant.
+    ///
+    /// Always return `true`.
+    #[deprecated]
```
Comment (kchibisov) [inline src/platform/wayland.rs:104]: Could we remove it entirely?

File: src/platform/x11.rs [hunk 3]
```diff
@@ -225,9 +128,8 @@ pub trait WindowExtUnix {
     fn is_ready(&self) -> bool;
```
Comment (kchibisov) [inline src/platform/x11.rs:128]: This should be removed as well?

**Review by kchibisov — CHANGES_REQUESTED** [review]

**Review by kchibisov — CHANGES_REQUESTED** [review]
CI.

**Review by kchibisov — APPROVED** [review]

**Pull request merged**